src/raftstore/store/fsm/apply.rs

```diff
@@ -2637,7 +2640,7 @@ impl ApplyFsm {
             .iter()
             .any(|res| res.region_id == self.delegate.region_id())
             && self.delegate.last_sync_apply_index != applied_index;
-        fail_point!("apply_on_handle_snapshot_sync", |_| { need_sync = true });
+        (|| fail_point!("apply_on_handle_snapshot_sync", |_| { need_sync = true }))();
```

Comment:
Why is this change needed?

Reply:
It's just the original code which was modified wrongly by #4903. Actually the tests should always fail without this revert, however there is a bug in our CI, which skips the fail cases.